sdk/storage/azure-storage-file-share/src/main/java/com/azure/storage/file/share/ShareFileAsyncClient.java

```diff
@@ -964,38 +964,33 @@ public Mono<Response<ShareFileProperties>> downloadToFileWithResponse(String dow
     Mono<Response<ShareFileProperties>> downloadToFileWithResponse(String downloadFilePath, ShareFileRange range,
         ShareRequestConditions requestConditions, Context context) {
         return Mono.using(() -> channelSetup(downloadFilePath, StandardOpenOption.WRITE, StandardOpenOption.CREATE_NEW),
-            channel -> getPropertiesWithResponse(requestConditions, context).flatMap(response ->
-                downloadResponseInChunk(response, channel, range, requestConditions, context)), this::channelCleanUp);
+            channel ->
+                downloadSingleChunk(channel, range, requestConditions, context), this::channelCleanUp);
     }
 
-    private Mono<Response<ShareFileProperties>> downloadResponseInChunk(Response<ShareFileProperties> response,
-        AsynchronousFileChannel channel, ShareFileRange range, ShareRequestConditions requestConditions,
-        Context context) {
-        return Mono.justOrEmpty(range).switchIfEmpty(Mono.defer(() -> Mono.just(new ShareFileRange(0, response.getValue()
-            .getContentLength()))))
-            .map(currentRange -> {
-                List<ShareFileRange> chunks = new ArrayList<>();
-                for (long pos = currentRange.getStart(); pos < currentRange.getEnd(); pos += FILE_DEFAULT_BLOCK_SIZE) {
-                    long count = FILE_DEFAULT_BLOCK_SIZE;
-                    if (pos + count > currentRange.getEnd()) {
-                        count = currentRange.getEnd() - pos;
-                    }
-                    chunks.add(new ShareFileRange(pos, pos + count - 1));
-                }
-                return chunks;
-            }).flatMapMany(Flux::fromIterable).flatMap(chunk ->
-                downloadWithResponse(new ShareFileDownloadOptions().setRange(chunk).setRangeContentMd5Requested(false)
-                    .setRequestConditions(requestConditions), context)
-                .map(ShareFileDownloadAsyncResponse::getValue)
-                .subscribeOn(Schedulers.boundedElastic())
-                .flatMap(fbb -> FluxUtil
-                    .writeFile(fbb, channel, chunk.getStart() - (range == null ? 0 : range.getStart()))
-                    .subscribeOn(Schedulers.boundedElastic())
-                    .retryWhen(Retry.max(3).filter(throwable -> throwable instanceof IOException
-                        || throwable instanceof TimeoutException))))
-            .then(Mono.just(response));
+
+    private static Mono<Void> writeBodyToFile(ShareFileDownloadAsyncResponse response, AsynchronousFileChannel channel,
```

Comment:
We may want to expose this through a new API, similar to what Blobs is doing where the Response type is something custom. Callers into the download without Response can call through this as well. That way we can offer a better choice but retain functionality of `ShareFileProperties`, and we won't need to add the translation code.